### PR TITLE
Support Asynchronous testing

### PR DESCRIPTION
**Pull request by bbalser**
Should address https://github.com/bbalser/Placebo/issues/13

### CODE REVIEW

File: test/async_test.exs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule Placebo.Async.Test1 do
+  use ExUnit.Case, async: true
+  use Placebo
+  require Logger
```
Comment (jeffgrunewald) [inline test/async_test.exs]: You don't seem to be using these requires.

**Review by jeffgrunewald — APPROVED** [review]

File: lib/placebo.ex [hunk 2]
```diff
@@ -121,24 +108,35 @@ defmodule Placebo do
...
       import Placebo.Actions
+      require Placebo.Macros
 
-      setup do
-        on_exit(fn ->
-          mocks = Placebo.Server.get()
+      setup(context) do
+        async? = Map.get(context, :async, false)
 
+        if async? == false do
```
Comment (jdenen) [inline lib/placebo.ex]: This is nitpicky, but can we change to `unless async?` or `if not async?`?

File: lib/placebo.ex [hunk 3]
```diff
@@ -121,24 +108,35 @@ defmodule Placebo do
...
-                       expectation.args
+                       stub.module,
+                       stub.function,
+                       stub.args,
+                       test_pid
                      )
             end)
           after
-            Placebo.Server.clear()
+            if async? == false do
```
Comment (jdenen) [inline lib/placebo.ex]: Same here.

File: lib/placebo.ex (hunk 2 above)
Comment (jdenen) [reply lib/placebo.ex]: I'm afraid this nitpicky comment is going to spawn an `IfServer` implementation for placebo, but I'm going to ask for this change anyway.

**Pull request merged**